src/updaters/update.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    This file contains methods for simulating change in a
+    landscape with expected amounts of spatial and temporal 
+    autocorrelation. 
+"""
+
+"""
+    NeutralLandscapeUpdater is an abstract type for methods
+    for updating a landscape matrix
+"""
+abstract type NeutralLandscapeUpdater end 
+
+"""
+    All `NeutralLandscapeUpdater`s have a field `rate`
+    which defines the expected change in any cell per timestep.  
+"""
+rate(up::NeutralLandscapeUpdater) = up.rate
+
+generator(up::NeutralLandscapeUpdater) = up.generator
+
+function update(updater::T, mat) where {T<:NeutralLandscapeUpdater}
+    _update(updater, mat)
+end
+@kwdef struct SpatiallyAutocorrelatedUpdater{G,R} <: NeutralLandscapeUpdater
+    generator::G = DiamondSquare(0.5)
+    rate::R = 0.1
+end
+function _update(sau::SpatiallyAutocorrelatedUpdater, mat)
+    change = rand(generator(sau), size(mat))
+    delta = rate(sau) .+ rate(sau) .* transform(fit(ZScoreTransform, change), change)
+    mat .+ delta
+end
+
+
+@kwdef struct SpatiotemporallyAutocorrelatedUpdater{G,R,T} <: NeutralLandscapeUpdater
+    generator::G = DiamondSquare(0.1)
+    rate::R = 0.1 
+    temporalupdater::T = BrownianMotion()
+end
+function _update(stau::SpatiotemporallyAutocorrelatedUpdater, mat)
+    change = rand(generator(stau), size(mat))
+    temporalshift = broadcast(x->update(stau.temporalupdater, x), mat)
+    z = transform(fit(ZScoreTransform, change), change)
+    delta = rate(stau) .+ (rate(stau) .* z) .+ (temporalshift .* z)
```

Comment:
Is there a reason this is on two lines? Instead of a single fused broadcast it will allocate an intermediate array. It could be:

```julia
mat .+ rate(stau) .+ (rate(stau) .* z) .+ (temporalshift .* z)
```

You could also include the `temporalshift` broadcast above in this line. It does lose a little readability.